### DISCUSSION TITLE
QT c++11support·报错

### DISCUSSION THREAD

**Discussion by 0KEAHA** [liusy58/NJU_AdvancedProgramming_SP22#178] [Q&A]
QT 项目编译时，出现300多个报错，都是系统文件的报错，在qbasicatomic.h中如下报错
![QQ截图20220425194322](https://user-images.githubusercontent.com/101489263/165082882-588f7947-477e-4bb1-9172-d3865a8873b0.jpg)
qbytearray.h中如下报错
![QQ截图20220425194848](https://user-images.githubusercontent.com/101489263/165083214-9823c493-624f-4d03-93b5-ea6f74e31f55.jpg)
在各种QT头文件中出现语法错误报错

在自己新建的头文件中,include map iostream vector等c++11文件均无报错,完全正常

Proj中已使用CONFIG += C++11，添加QMAKE_CXXFLAGS += -std=c++11依然不能解决问题
系统path如图
Qt版本是5.14.2，此前曾安装5.9版本后使用qt自带的卸载工具将其卸载
项目曾多次成功编译过，在实现了自建非QObject类，使用cstdlib头文件以后编译发现问题，尚不清楚是否有关

![QQ截图20220425194608](https://user-images.githubusercontent.com/101489263/165082825-4f50a206-4b41-4f55-9dff-0f70ee3c2d4f.jpg)
![QQ截图20220425194618](https://user-images.githubusercontent.com/101489263/165082846-82f993d0-e551-4383-a8ac-fbb0a83a2cec.jpg)

将no complier修改为msvc 等等后不能解决问题，且界面中的三角警告转为红色错误

重装了qt5.14.2，安装gcc6.3 以后，未发生任何改变

 后来尝试运行了QT官方提供的示例，示例运行正常，将自己的项目中两个未涉及QT操作的cpp和h文件remove后，编译通过，出现了wedget窗口

后将这个类文件加入以后，将内部代码注释，再次编译通过

类文件使用了静态变量，在cpp文件中初始化以后，无任何红色错误

事实上，在remove文件并重新添加以后，原先的300多条错误均消失。

虽然问题已经解决，但无法得知原因，下次遇到类似情况，尝试将项目文件全部移除再重新添加可能有所帮助。

**Comment by liusy58** [comment]
恭喜进入计算机的魔法世界！如果有找到原因欢迎追更～（开个玩笑

有很多时候发现莫名报错的话可以尝试关机重启，如果你认为你不能找到原因可以来找我们讨论，不过最好约一个时间～
